tests/foreman/cli/test_contentviewfilter.py

```diff
@@ -0,0 +1,431 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Content View Filters"""
+
+from ddt import ddt
+from fauxfactory import gen_choice, gen_string
+from random import randint
+from robottelo.cli.contentview import ContentView
+from robottelo.cli.factory import (
+    CLIFactoryError,
+    make_content_view,
+    make_org,
+    make_product,
+    make_repository,
+)
+from robottelo.cli.repository import Repository
+from robottelo.common.constants import DOCKER_REGISTRY_HUB
+from robottelo.common.decorators import data, skip_if_bug_open
+from robottelo.test import CLITestCase
+
+
+@ddt
+class TestContentViewFilter(CLITestCase):
+    """Content View Filter CLI tests"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Init single organization, product and repository for all tests"""
+        super(TestContentViewFilter, cls).setUpClass()
+        cls.org = make_org()
+        cls.product = make_product({u'organization-id': cls.org['id']})
+        cls.repo = make_repository({u'product-id': cls.product['id']})
+        Repository.synchronize({u'id': cls.repo['id']})
+
+    def setUp(self):
+        """Init content view with repo per each test"""
+        super(TestContentViewFilter, self).setUp()
+        self.cvf_name = gen_string('utf8')
+        try:
+            self.content_view = make_content_view({
+                u'organization-id': self.org['id'],
+            })
+        except CLIFactoryError as err:
+            self.fail(err)
+
+        result = ContentView.add_repository({
+            u'id': self.content_view['id'],
+            u'repository-id': self.repo['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+
+    @data(
+        gen_string('alphanumeric', randint(1, 255)),
+        gen_string('alpha', randint(1, 255)),
+        gen_string('cjk', randint(1, 85)),
+        gen_string('latin1', randint(1, 255)),
+        gen_string('numeric', randint(1, 255)),
+        gen_string('utf8', randint(1, 85)),
+        gen_string('html', 85),
+    )
+    def test_create_cvf_with_different_names(self, name):
+        """Test: Create new content view filter and assign it to existing
+        content view by id. Use different value types as a name and random
+        filter content type and random filter type as a parameters for this
+        filter
+
+        @Feature: Content View Filter
+
+        @Assert: Content view filter created successfully and has correct and
+        expected parameters
+
+        """
+        filter_content_type = gen_choice(['rpm', 'package_group', 'erratum'])
```

Comment:
@oshtaier I take it that `test_create_cvf_with_content_types` addresses this comment?